src/docs/release/breaking-changes/12_18_19_ImageCache_Logic.md

```diff
@@ -0,0 +1,69 @@
+---
+title: Revert ImageCache algorithm for large image
+description: Stop increasing the ImageCache maxByteSize to accomodate large
+images.
+---
+
+# Revert ImageCache algorithm for large image
+
+## Context
+
+When loading images into the ImageCache whose byte size is larger than the
+ImageCache's maxByteSize we previously would permanently increase the
+maxByteSize to accommodate them.  This logic can lead to creeping maxByteSize
+which makes working in memory limited systems more difficult.
+
+## Description of change
+
+The following changes were made to the ImageCache algorithm:
+
+```dart
+// Old Logic Pseudocode
+void onLoadImage(Image image) {
+  if (image.byteSize > _cache.maxByteSize) {
+    _cache.maxByteSize = image.byteSize + 1000;
+  }
+  _cache.add(image);
+  while (_cache.count > _cache.maxCount 
+      || _cache.byteSize > _cache.maxByteSize) {
+    _cache.discardOldestImage();
+  }
+}
+```
+
+```dart
+// New Logic Pseudocode
+void onLoadImage(Image image) {
+  if (image.byteSize < _cache.maxByteSize) {
+    _cache.add(image);
+    while (_cache.count > _cache.maxCount 
+        || _cache.byteSize > cache.maxByteSize) {
+      cache.discardOldestImage();
+    }
+  }
+}
+```
+
+## Migration guide
+
+There may be situations where the ImageCache is thrashing with the new logic
+where it wasn't previously, specifically if you are loading images that are
+larger than your cache.maxByteSize.  These can be remedied by one of the
+following:
+
+  1) Increase the ImageCache.maxByteSize to accommodate the large images.
+  1) Adjust your image loading logic to guarantee that the images will fit
+     nicely into the ImageCache.maxByteSize of your choosing.
+  1) Subclass ImageCache, implement your desired logic, create a new bindings
+     that will serve up your subclass of ImageCache.
```

Comment:
It would help to have an example of how to do this here - where in a Flutter program should you set the cache, etc.

Reply:
I have an outstanding issue to add that documentation.  I don't think there is a good place to point people yet today: https://github.com/flutter/flutter/issues/47388

Reply:
We make this harder than I realized.  Nevermind.

Reply:
Can you add a link in here to the example you're creating at https://github.com/flutter/flutter/pull/47467?  It can probably just be a link to the `PictureCache` class doc.

Reply:
Done.